### ISSUE TITLE
Feature Request: Support for `[Searchable]`

### ISSUE THREAD

**Issue by tim-altitudegames** [TylerTemp/SaintsField#24]
Something I'd love to see is support for the `[Searchable]` attribute for lists/arrays, helps deal with sizeable entities in the inspector

![329103583-e1872856-85cf-405d-bbf1-2957bbb9d04e](https://github.com/TylerTemp/SaintsField/assets/150100676/40a74c23-ad0f-4b3e-b79e-69bbc946ea60)

**Comment by TylerTemp** [comment]
Unfortunately, this feature requires some heavy work.

For most of the time, you can always use Odin together with SaintsField, as long as you
1.  Do not enable SaintsEditor globally (and not use SaintsEditor related properties)
2.  Do not use some edge cases like Odin's serialization tool for non-serializable object, especially Odin's EditorWindow.

The difficulty of this feature is, Unity does not naturally allow to make a "list/array drawer" with a `CustomPropertyDrawer` way, but only allows on `UnityEditor.Editor` way. Which means this feature requires to completely re-write Unity's reorder-able list rendering. Some other workaround is to hijack the property drawer for first element of array/list and ignore the rest element (like how [Editor GUI Table](https://assetstore.unity.com/publishers/34298) did)

So, in short, I'll only re-consider it again after I finish some other features for this project. Before that I'll leave this issue open for now.

Any suggestion & PR is welcome!

**Comment by TylerTemp** [comment]
@tim-altitudegames Hi,

Please check [version 3.1.1](https://github.com/TylerTemp/SaintsField/releases/tag/3.1.1) and [ListDrawerSettings document](https://github.com/TylerTemp/SaintsField/tree/3.1.1?tab=readme-ov-file#listdrawersettings)

This issue is closed as implemented. However, this component is quite heavy so is expected to have some bugs. Feel free to report issues if you find any problems.

```csharp
using SaintsField.Playa;

[Serializable]
public struct MyData
{
    public int myInt;
    public string myString;
    public GameObject myGameObject;
    public string[] myStrings;
}

[ListDrawerSettings(searchable: true, numberOfItemsPerPage: 3)]
public MyData[] myDataArr;
```

![image](https://github.com/TylerTemp/SaintsField/assets/6391063/08c6da9a-e613-4e94-8933-3d7a92f7cb33)
